### PR TITLE
Add Localhost Demo CI to GitHub workflows

### PR DESCRIPTION
**Pull request by jasonpcarroll**
*Issue #, if available:*

*Description of changes:*
This adds workflows to build and execute non-aws demos on pull requests.

DO NOT APPROVE until https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/59 is merged.

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: .github/workflows/aws_demo_runner.yml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        run: sudo apt-get install -y libmosquitto-dev
+
+      - name: Install CMake
+        run: |
+          curl https://cmake.org/files/v3.2/cmake-3.2.0-Linux-x86_64.tar.gz -o cmake.tar.gz
+          tar -xf cmake.tar.gz
+
+      - name: Generate SSL credentials
+        id: generate-credentials
+        uses: jasonpcarroll/CI-CD-GitHub-Actions/ssl-credential-creator@main
```
Comment (Skptak) [inline .github/workflows/aws_demo_runner.yml:35]: There are a couple places where you have this pointing to your local fork. Can you update these to point to the FreeRTOS repo?

File: .github/workflows/aws_demo_runner.yml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+          -DCLAIM_CERT_PATH="cert/path" \
+          -DCLAIM_PRIVATE_KEY_PATH="key/path" \
+          -DPROVISIONING_TEMPLATE_NAME="template-name" \
+          -DDEVICE_SERIAL_NUMBER="00000" \
+          -DCSR_SUBJECT_NAME="CN=Fleet Provisioning Demo" \
+          -DGREENGRASS_ADDRESS="greengrass-address"
+
+      - name: Configure the HTTP Plaintext Demo
+        working-directory: demos/http/http_demo_plaintext
+        run: sed -i '1s/^/#define HTTP_PORT 8080\n/' demo_config.h
```
Comment (Skptak) [inline .github/workflows/aws_demo_runner.yml:75]: Kind of a nit, but maybe a sed command more like `sed -i 's/define HTTP_PORT[[:blank:]]\{1,\}( 80 )/define HTTP_PORT ( 8080 )/g' demo_config.h` that way there aren't two values being defined for if searching through logs?

File: .github/workflows/aws_demo_runner.yml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+          server-cert-path: ${{ steps.generate-credentials.outputs.server-cert-path }}
+
+      - name: Start localhost HTTP server
+        id: http-server
+        uses: Eun/http-server-action@v1.0.6
+
+      - name: Configure CMake Build for Localhost servers
+        run: |
+          mkdir build && cd build
+          ../cmake-3.2.0-Linux-x86_64/bin/cmake .. \
```
Comment (Skptak) [inline .github/workflows/aws_demo_runner.yml:52]: Is there a reason that the specific version of cmake needs to be specified like this? Seems like this is something that should just be able to be called usign just "cmake" from the path

File: .github/workflows/aws_demo_runner.yml [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        run: cd build && make http_demo_plaintext
+
+      - name: Run and monitor the HTTP Plaintext Demo
+        continue-on-error: true
+        uses: jasonpcarroll/CI-CD-GitHub-Actions/executable-monitor@main
+        with:
+          exe-path: ./build/bin/http_demo_plaintext
+          log-dir: demo_run_logs
+          timeout-seconds: 300
+          success-line: "Demo completed successfully."
```
[the remark this answers is not in the document]
Comment (Skptak) [reply .github/workflows/aws_demo_runner.yml:90]: In my mind this extends to all these default values being passed in here

[the remark this answers is not in the document]
Comment (jasonpcarroll) [reply .github/workflows/aws_demo_runner.yml:90]: These are all input variables - not hard coded - to the action :)

[the remark this answers is not in the document]
Comment (jasonpcarroll) [reply .github/workflows/aws_demo_runner.yml:90]: You can specify on a demo to demo basis

File: .github/workflows/aws_demo_runner.yml (hunk 2 above)
Comment (jasonpcarroll) [reply .github/workflows/aws_demo_runner.yml]: Done and I updated all other sed commands as well.

File: .github/workflows/aws_demo_runner.yml (hunk 3 above)
Comment (jasonpcarroll) [reply .github/workflows/aws_demo_runner.yml]: Updated to use an action from the github marketplace

**Review by Skptak — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**